drakrun/drakrun/draksetup.py

```diff
@@ -390,44 +392,38 @@ def send_usage_report(report):
-        mnt_path_quoted = shlex.quote(mount_path)
-        blk_quoted = shlex.quote(block_device)
+def create_rekall_profiles(install_info: InstallInfo, runtime_info: RuntimeInfo, kernel_profile: str):
+    injector = Injector('vm-0', runtime_info, kernel_profile)
+
+    for file in dll_file_list:
         try:
-            subprocess.check_output(f"mount -t ntfs -o ro {blk_quoted} {mnt_path_quoted}", shell=True)
-        except subprocess.CalledProcessError:
-            raise RuntimeError(f"Failed to mount {block_device} as NTFS.")
+            logging.info(f"Fetching rekall profile for {file.path}")
 
-        for file in dll_file_list:
-            try:
-                logging.info(f"Fetching rekall profile for {file.path}")
-                local_dll_path = os.path.join(PROFILE_DIR, file.dest)
-
-                copyfile(os.path.join(mount_path, file.path), local_dll_path)
-                guid = pdb_guid(local_dll_path)
-                tmp = fetch_pdb(guid["filename"], guid["GUID"], PROFILE_DIR)
-
-                logging.debug("Parsing PDB into JSON profile...")
-                profile = make_pdb_profile(tmp)
-                with open(os.path.join(PROFILE_DIR, f"{file.dest}.json"), 'w') as f:
-                    f.write(profile)
-            except FileNotFoundError:
-                logging.warning(f"Failed to copy file {file.path}, skipping...")
-            except RuntimeError:
-                logging.warning(f"Failed to fetch profile for {file.path}, skipping...")
-            except Exception:
-                logging.warning(f"Unexpected exception while creating rekall profile for {file.path}, skipping...")
-            finally:
-                if os.path.exists(local_dll_path):
-                    os.remove(local_dll_path)
-                if os.path.exists(os.path.join(PROFILE_DIR, tmp)):
-                    os.remove(os.path.join(PROFILE_DIR, tmp))
-
-        # cleanup
-        subprocess.check_output(f'umount {mnt_path_quoted}', shell=True)
+            local_dll_path = os.path.join(PROFILE_DIR, file.dest)
+            guest_dll_path = os.path.join("C:", file.path)
+
+            injector.read_file(guest_dll_path, local_dll_path)
+            guid = pdb_guid(local_dll_path)
+            tmp = fetch_pdb(guid["filename"], guid["GUID"], PROFILE_DIR)
+
+            logging.debug("Parsing PDB into JSON profile...")
+            profile = make_pdb_profile(tmp)
+            with open(os.path.join(PROFILE_DIR, f"{file.dest}.json"), 'w') as f:
+                f.write(profile)
+        except subprocess.CalledProcessError:
+            logging.warning(f"Failed to copy file {file.path}, skipping...")
+        except RuntimeError:
+            logging.warning(f"Failed to fetch profile for {file.path}, skipping...")
+        except Exception:
+            # Can help in debugging
+            traceback.print_exc()
+            logging.warning(f"Unexpected exception while creating rekall profile for {file.path}, skipping...")
```

Comment:
I can add a `[DEBUG]` tag before these debug tracebacks and show the warning above the traceback to avoid the confusions if it seems better
```suggestion
            logging.warning(f"Unexpected exception while creating rekall profile for {file.path}, skipping...")
            logging.debug(traceback.format_exc())
```

Giving the output like this,
```
[2021-03-14 19:45:32,409][DEBUG] Parsing PDB into JSON profile...   
[2021-03-14 19:45:46,045][WARNING] Unexpected exception while creating rekall profile for Windows/SysWOW64/ole32.dll, skipping...                                                                                                                                  
[2021-03-14 19:45:46,045][DEBUG] Traceback (most recent call last):                                                                                                                           
  File "/home/user/drakvuf-sandbox/drakrun/drakrun/draksetup.py", line 410, in create_rekall_profiles                                                                                         
    profile = make_pdb_profile(tmp)                                                                                                                                                           
  File "/home/user/drakvuf-sandbox/drakrun/drakrun/drakpdb.py", line 241, in make_pdb_profile                                                                                                 
    pdb = pdbparse.parse(filepath)                                                                                                                                                            
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/__init__.py", line 554, in parse                                                                         
    return PDB7(f, fast_load)
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/__init__.py", line 521, in __init__                                                                      
    self.read_root(self.root_stream)
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/__init__.py", line 467, in read_root                                                                     
    parent = self))
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/__init__.py", line 154, in __init__                                                                      
    self.load()
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/__init__.py", line 262, in load                                                                          
    tpis = tpi.parse_stream(self.stream_file, unnamed_hack, elim_fwdrefs)
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/tpi.py", line 1190, in parse_stream                                                                      
    types[i].substructs = ListContainer([resolve_typerefs(t, types, min) for t in types[i].substructs])                                                                                      
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/tpi.py", line 1190, in <listcomp>                                                                        
    types[i].substructs = ListContainer([resolve_typerefs(t, types, min) for t in types[i].substructs])                                                                                      
  File "/home/user/drakvuf-sandbox/drakrun/env/lib/python3.7/site-packages/pdbparse/tpi.py", line 1105, in resolve_typerefs                                                                  
    for attr in type_refs[leaf.leaf_type]:
KeyError: 62195
                                                  
[2021-03-14 19:45:46,048][INFO] Fetching rekall profile for Windows/System32/combase.dll

```

Reply:
So I understand that this error also appears for `combase.dll` in current version of Sandbox, but it's simply silenced?

Reply:
I don't really have track for what all DLLs this error occurs but yes, This was silenced before when the exception was handled

Reply:
I modified the code a bit locally to capture the DLLs categorically which cause errors, here are the results
- `Error DLLs` show the traceback which you had seen

Other names should be intuitive ig
```
[2021-03-15 02:07:00,487][DEBUG] Error DLLs: ['Windows/System32/ole32.dll', 'Windows/SysWOW64/ole32.dll']
[2021-03-15 02:07:00,490][DEBUG] runtime error DLLs: []
[2021-03-15 02:07:00,490][DEBUG] failed_to_copy DLLs: ['Windows/System32/combase.dll']

```

Reply:
Cool!